File: leetcode/easy/java/LengthOfLastWord.java
package leetcode.easy.java;

// Given a string s consisting of words and spaces, return the length of the last word in the string.
// * A word is a maximal substring consisting of non-space characters only.


class LengthOfLastWord {
    public int lengthOfLastWord(String s) {
        String[] words = s.trim().split("\\s+");  
        return words[words.length - 1].length();              
    }
}

// s.trim():       trims any leading or trailing whitespace from the input string
// split("\\s+"):  uses a regular expression to split the string around sequences of one or more whitespace characters. It results in an array of words
//                 \  escape
//                 \s matches any whitespace character
//                 +  matches one or more of the preceding element
// words[words.length - 1]: accesses the last element in the array, which is the last word

File: leetcode/easy/python/lengthOfLastWord.py
# Given a string s consisting of words and spaces, return the length of the last word in the string.
# * A word is a maximal substring consisting of non-space characters only.


class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        words = s.strip().split()
        return len(words[-1])


# Strip any leading or trailing spaces, then split the string by whitespace
# Return the length of the last word in the list
